optimum/intel/openvino/modeling_base.py

```diff
@@ -73,9 +73,19 @@ def __init__(
         self.is_dynamic = dynamic_shapes
         self.ov_config = {} if ov_config is None else {**ov_config}
         self.preprocessors = kwargs.get("preprocessors", [])
+        self.compile_only = kwargs.get("compile_only", False)
         enable_compilation = kwargs.get("compile", True)
 
-        if self.is_dynamic:
+        if self.compile_only and not enable_compilation:
+            raise ValueError(
+                "`compile_only` mode does not support disabling compilation."
+                "Please provide `compile=True` if you want to use `compile_only=True` or set `compile_only=False`"
+            )
+
+        if self.compile_only and not isinstance(model, CompiledModel):
+            raise ValueError("`compile_only` expect that already compiled model will be provided")
+
+        if self.is_dynamic and not self.compile_only:
```

Comment:
Shouldn't we set `self.is_dynamic` to False if `self._compile_only=True` ?

something like 
```suggestion
        if self.is_dynamic:
```

and 

```
self.is_dynamic = dynamic_shapes and not self.compile_only
```